### ISSUE TITLE
globalShortcut can't register OS shortcuts

### ISSUE THREAD

**Issue by fcastilloec** [electron/electron#10159]
* Electron version: 1.6.11
* Operating system: Ubuntu 16.04 & 17.04. Elementary OS Loki, and other linux systems

### Expected behavior

I should be able to receive a MediaPlayPause key event on my app

### Actual behavior

Registration fails for the shortcuts

### How to reproduce

you can clone https://github.com/electron/electron-quick-start and add these lines to createWindow function
```
var playpause = globalShortcut.register('MediaPlayPause', () => {
    console.log('MediaPlayPause is pressed')
  });
  
  if (!playpause) {
    console.log('MediaPlayPause registration failed');
  } else {
    console.log('MediaPlayPause registration worked');
  }
```
-----------------------------
So, my OS does have registered the MediaPlayPause and that might be why it's not being registered, but how else can I pass a global OS shortcut to my app? If my app is a music player, I should be able to control it with the OS global shortcuts, which my app should be able to listen.

**Comment by MarshallOfSound** [comment]
GitHub issues are for feature requests and bug reports, questions about using Electron or code assistance requests should be directed to the [community](https://github.com/electron/electron#community) or to the [Slack Channel](http://atom-slack.herokuapp.com/).

If an existing Chromium / Electron app has registered the media keys as hotkeys then you will not be able to register them.  Ensure the no Chrome extension is using them and you don't have any other Electron music players running.

**Comment by fcastilloec** [comment]
This code has been run when no other application was running a the time on a fresh Ubuntu install. The problem lies that electron can't receive OS shortcuts (like media related) on its apps.
You should first try the code I've posted on Ubuntu, with no other apps running.
The only way to fix this issue is if I change my OS key bindings, so the electron app works properly. This also means that any other audio/music player will have to use the new shortcuts, while electron will be the only one using the media keys.

**Comment by MarshallOfSound** [comment]
> The problem lies that electron can't receive OS shortcuts (like media related) on its apps.

I know this is incorrect because I use the media shortcuts in my apps on all three platforms and don't have any issues.  I tried your code on my ubuntu VM and it appears to work fine?

**Comment by jostrander** [comment]
The reason the media keys work on linux in gpmdp is because I wrote the code to integrate with the media service in gnome and other services. They actually take exclusive control of the media buttons at system start and hand out the events to the currently "selected" application, @MarshallOfSound.

**Comment by MarshallOfSound** [comment]
Ah yeah, woops, forgot we have that DBus stuff

@fcastilloec that logic is probably what you're looking for.  You may want to look at the `mpris` stuff there as well 👍

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!

**Comment by fcastilloec** [comment]
Just wanted to say that this bug has been fixed, tested with electron `v3.0.2`. I was using the same method as described in https://github.com/electron/electron/issues/10159#issuecomment-319556609, with dbus, and it's been working great so far, but I rather use the default `globalShortcut` module to make the code work better across all OSs

**Comment by codebytere** [comment]
@fcastilloec awesome! thanks for letting us know 😁

**Comment by gadtfly** [comment]
@fcastilloec: can you clarify what you meant in https://github.com/electron/electron/issues/10159#issuecomment-425675448?

Did you mean that you've found that Electron has fixed the problem in v3.0.2?  Or did you mean that you fixed the problem in your *own* code, using the DBus workaround from https://github.com/electron/electron/issues/10159#issuecomment-319556609?

Because I am still experiencing this problem in electron v4.1.0.

**Comment by fcastilloec** [comment]
@gadtfly Sorry, I think I didn't do a proper test of my code when I posted that. I don't think the bug has been fixed. I have two parts in my code, one for shortcuts and another for MPRIS. I used to use dbus for shortcuts as my workaround. Moder Linux OSs, especially if they're using Gnome3, send shortcut signals using MPRIS. So, when I tested my code, I thought shortcuts were working but in reality, I don't need any shortcut code anymore, just my MPRIS code. If I didn't have MPRIS, then shortcuts wouldn't work.
